x/params/contents.go

```diff
@@ -0,0 +1,85 @@
+package params
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov/proposal"
+)
+
+const RouterKey = "params"
+
+// Struct for single parameter change
+type Change struct {
+	Space  string `json:"space"`
+	Key    []byte `json:"key"`
+	Subkey []byte `json:"subkey"`
+	Value  []byte `json:"value"`
+}
+
+// Constructs new Change
+func NewChange(space string, key, subkey, value []byte) Change {
+	return Change{space, key, subkey, value}
+}
+
+func (c Change) String() string {
+	var subkey string
+	if len(c.Subkey) != 0 {
+		subkey = "(" + string(c.Subkey) + ")"
+	}
+	return fmt.Sprintf("{%s%s := %X} (%s)", string(c.Key), subkey, c.Value, c.Space)
+}
+
+// ValidateChange checks whether the input data is empty or not
+func ValidateChanges(changes []Change) sdk.Error {
+	if len(changes) == 0 {
+		return ErrEmptyChanges(DefaultCodespace)
+	}
+
+	for _, c := range changes {
+		if len(c.Space) == 0 {
+			return ErrEmptySpace(DefaultCodespace)
+		}
+		if len(c.Key) == 0 {
+			return ErrEmptyKey(DefaultCodespace)
+		}
+		if len(c.Value) == 0 {
+			return ErrEmptyValue(DefaultCodespace)
+		}
+	}
+	return nil
+}
+
+// Proposal which contains multiple changes on proposals
+type ChangeProposal struct {
```

Comment:
ditto

Reply:
also this should be `ParameterChangeProposal`

Reply:
It will be `params.ParameterChangeProposal`.. but I agree it makes more sense to name it `ParameterChangeProposal`

Reply:
Is `ParamChangeProposal` looks good?